services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +451,24 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(batch []map[string]interface{}, timestamp time.Time, p producerManager) (
```

Comment:
Can we returned message batches grouped by topic? Even if transformer might do something similar, it won't hurt doing it here. In the worst case we will be returning only 1 group.
see

Reply:
I was under the impression that this was not needed. Please @utsabc clarify what it's done on the transformer side. Can you provide a full example with a data sample of an input coming to the transformer and the output produced by the transformer?

Reply:
Yes sure this is the [function](https://github.com/rudderlabs/rudder-transformer/blob/f76b4c18825fd94116dfba21b9e9fc213c1d8e71/v0/destinations/kafka/transform.js#L22) where during batch phase we are receiving events form server and we are grouping them by topic. The batched request for all events having same topics are sent as one request from transformer, we did this so that failure in batch for a topic has no side effects on events for another topic in batch scenario

This is an input test [case](https://github.com/rudderlabs/rudder-transformer/blob/master/__tests__/data/kafka_batch_input.json) for batch function I shared above 
and this is the [output](https://github.com/rudderlabs/rudder-transformer/blob/master/__tests__/data/kafka_batch_output.json)
Essentially this is our unit test case for batch scneario

Reply:
Looks good to me, it looks like you used different topics as well in your test. @atzoum any thoughts?